File: 3705-find-the-largest-almost-missing-integer/find-the-largest-almost-missing-integer.java
class Solution {
    public int largestInteger(int[] nums, int k) {
        int[] freq = new int[51];
        int ans = Integer.MIN_VALUE;
        if(k == nums.length){
            for(int i = 0; i < nums.length; i++){
                ans = Math.max(ans, nums[i]);
            }
            return ans;
        }

        int i = 0;
        while(i + k <= nums.length){
            for(int j = i; j < k + i; j++){
                freq[nums[j]]++;
            }
            i++;
        }

        for(int j = freq.length - 1; j >= 0; j--){
            if(freq[j] == 1){
                return j;
            }
        }

        return -1;
    }
}

File: 3705-find-the-largest-almost-missing-integer/find-the-largest-almost-missing-integer.py
class Solution:
    def largestInteger(self, nums: List[int], k: int) -> int:
        a=nums[0]
        b=nums[-1]
        c=[-1]
        if len(nums)==k:
            return max(nums)
        if k==1:
            for i in nums:
                if nums.count(i)==1:
                    c.append(i)
            return max(c)
        if nums.count(a)==1:
            c.append(a)
        if nums.count(b)==1:
            c.append(b)
        return max(c)
